tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -3479,7 +3489,20 @@ def _to_proto(self):
     pb.optimization_options.CopyFrom(self.experimental_optimization._to_proto())  # pylint: disable=protected-access
     if self.experimental_slack is not None:
       pb.slack = self.experimental_slack
-    pb.threading_options.CopyFrom(self.experimental_threading._to_proto())  # pylint: disable=protected-access
+    # (kvignesh1420): We try to keep the values of `threading` and
+    # `experimental_threading` the same, to prevent unexpected behaviours
+    # and ensure backward-compatibility.
+    override_attrs = []
```

Comment:
instead of collecting a list of overridden and then printing a single warning, log the warning for each attribute, listing the divergent values:

```
for attr in filter(lambda opt: not opt.startswith("_"), dir(self.threading)):
  x, y = getattr(self.threading, attr), getattr(self.experimental_threading, attr)
  if (x != y):
    log.warning("overriding `experimental_threading.{} = {}` with `threading.{} = {}", attr, x, attr, y)
```

Reply:
addressed in the commit